src/jquery.number-input-formatter.js

```diff
@@ -0,0 +1,92 @@
+(function ($) {
+
+  function numberInputFormatter(options) {
+    options = options || {};
+    options.labelSelector = options.labelSelector || 'span';
+    $(this).on('keyup change blur focus', function (e) {
+      formatInput($(e.target), options.labelSelector)
+    });
+    $(this).each(function () {
+      formatInput($(this), options.labelSelector);
+    });
+    return this;
+  }
+
+  function formatInput($field, labelSelector) {
+    var value = $field.val();
+    var $label = getLabelFromField($field, labelSelector);
+    $label.text(formatNumber(value));
+  }
+
+  /**
+   * Get the sibling to write the formatted value to.
+   *
+   * @param jQuery $field The input
+   * @param String labelSelector A CSS selector
+   * @return jQuery
+   */
+  function getLabelFromField($field, labelSelector) {
+    return $field.next(labelSelector);
+  }
+
+  /**
+   * Format a number without providing the number of decimal places.
+   *
+   * This is called "simple" because it will automatically count the number of
+   * decimal places in the provided number.
+   *
+   * @param Number|String number The number to format
+   * @param String [decimal] The decimal separator, a period by default
+   * @param String [separator] The places separator, a comma by default
+   * @return String A properly formatted number
+   */
+  function formatNumber(number, decimal, separator) {
+    var decimals = countDecimals(number, decimal);
+    return formatNumberBase(number, decimals, decimal, separator);
+  }
+
+  /**
+   * Count the number of decimal places in a number.
+   *
+   * e.g.:
+   * 1234 returns 0
+   * 1234.56 returns 2
+   *
+   * @param Number|String number The number with decimal places to count
+   * @param String [decimal] The decimal separator, a period by default
+   * @return Number
+   */
+  function countDecimals(number, decimal) {
+    decimal = decimal || '.';
+    number = ('' + number).split(decimal);
+    return number[1] ? number[1].length : 0;
+  }
+
+  /**
+   * Number formatter.
+   *
+   * Inspired by http://stackoverflow.com/questions/149055/how-can-i-format-numbers-as-money-in-javascript
+   *
+   * n = amount
+   * c = # digits to the right of the decimal
+   * d = decimal (defaults to period)
+   * t = separator (defaults to comma)
+   * @return String  formatted number
+   */
+  function formatNumberBase(n, c, d, t) {
```

Comment:
Just 'cause I gotta ask (not necess that I think it's the greatest idea) could depending on numeral.js here help at all? Feels like this formatting is a shared-library problem whose maintenance could be pushed off to someone else. (Alternatively, this might be all it ever needs to do and that's a stupid reason to adopt a dependency... point of discussion).